extension/src/setup/index.ts

```diff
@@ -378,9 +378,18 @@ export class Setup
     const commandArgs = args.length === 0 ? '' : ` ${args.join(' ')}`
     const command = executable + commandArgs
 
+    if (!cwd) {
+      return {
+        command,
+        version: undefined
+      }
+    }
+
+    const version = await this.getCliVersion(cwd)
+
     return {
       command,
-      version: cwd ? await this.getCliVersion(cwd) : undefined
+      version: version || (await this.getCliVersion(cwd, true))
```

Comment:
Looking at the [`discovery.ts` code](https://github.com/iterative/vscode-dvc/blob/main/extension/src/cli/dvc/discovery.ts#L243), we check for the version, then check for a global version. Our function that gets DVC CLI details, however, wasn't checking for a global. 

This could possibly be why Ivan was seeing `Version: Not Found` but the section was saying `DVC Incompatible`.

Reply:
Why not save the version when setup gets run instead of recreating the logic here?

Reply:
I had a quick look at the code and potentially `setCliCompatible` in `Setup` could be passed the version/undefined, it could then set both the version and `cliCompatible`. The version can then be passed to the webview without having to call `this.getCliVersion` each time.

Reply:
I assume the current behaviour is why we are calling `dvc version ` all of the time too:


https://user-images.githubusercontent.com/37993418/235392775-7ec40861-df6c-48bb-89ea-eea625558d68.mov

Reply:
Thanks for the ideas, Matt! Updated the code to use a variable for version.